src/app/page.tsx

```diff
@@ -14,12 +14,34 @@ import { Tabs, TabsContent, TabsList, TabsTrigger } from '@/components/ui/tabs'
 import { getAllApplications } from '@/lib/apiClient'
 import { Search } from 'lucide-react'
 import { useQuery } from 'react-query'
+import { useState } from 'react'
+import Fuse from 'fuse.js'
 
 export default function Home() {
   const { data, isLoading, error } = useQuery({
     queryKey: ['application'],
     queryFn: getAllApplications,
   })
+  const [filter, setFilter] = useState('all')
+  const [searchTerm, setSearchTerm] = useState('')
+  const filteredData =
+    data?.filter(
+      (app) =>
+        filter === 'all' || app.info.application_lifecycle.state === filter,
+    ) || []
+
+  const fuse = new Fuse(filteredData, {
+    keys: [
```

Comment:
I would get the keys from one object and put them here, for example: `Object.keys(data[0])`
you need to make sure data[0] is exist though

Reply:
I dont see also where fuse is being triggered. 
maybe add `useEffect` that only runs when `searchTerm` is changed. also when you do search/filter you need to take into account the other parameter, ie:
when there is search, u need to also if any filters are set and when a filter is changed u need to also check if there is a search term